### ISSUE TITLE
[0.43.11-RC5 : Bug] Fatal error when "Restore Unit" Black Eagle

### ISSUE THREAD

**Issue by SirMegaV** [MegaMek/mekhq#748]
### Environment

0.43.11-RC5

### Description

Fatal error when "Restore Unit" Black Eagle. Game will be frozen. Nothing in log. Might be related to case #745 and #747. To replicate:

1. Load campaign
2. Go to hanger
3. Right click, GM Mode, Restore Unit
4. Game frozen

![image](https://user-images.githubusercontent.com/17064665/41187058-eb6b8288-6bd4-11e8-9d30-1b81b99d3231.png)

### Files

[Fatal error when Restore Unit Black Eagle.zip](https://github.com/MegaMek/mekhq/files/2086588/Fatal.error.when.Restore.Unit.Black.Eagle.zip)

**Comment by neoancient** [comment]
The InvocationTargetException in NativeMethodAccessorImpl.invoke0 is a wrapper around a different exception thrown in the process of dispatching an event. The cause is the exception thrown by the event listener, which in this case is a NullPointerException when attempting to update the daily log. The stack trace from the original exception is truncated, but this is what is in the log:

```Caused by: java.lang.NullPointerException
	at mekhq.gui.DailyReportLogPanel.appendLog(DailyReportLogPanel.java:112)
	at mekhq.gui.CampaignGUI.refreshReport(CampaignGUI.java:2969)
	at mekhq.gui.CampaignGUI.handle(CampaignGUI.java:3029)
	... 48 more```

I was unable to replicate it and there is nothing at that point in the code that suggests a cause to me.

**Comment by SirMegaV** [comment]
I tested once again, using the attached campaign file. Same error. The screen frozen, no log messages. 

What I did.
1. Download and unzip attached campaign
2. Run RC5, load campaign
3. Go to hanger tab
4. Right click Black Eagle, GM Mode, Restore Unit
5. Screen frozen

Log attached again.

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/2092928/mekhqlog.txt)

**Comment by NickAragua** [comment]
Got it. Looks like some kind of endless loop.

**Comment by NickAragua** [comment]
Fixed with #751
